services/src/main/java/org/fao/geonet/api/registries/vocabularies/KeywordsApi.java

```diff
@@ -308,7 +308,135 @@
         return searcher.getResults();
     }
 
-
+    /**
+     * Search keywords.
+     *
+     * @param q the q
+     * @param lang the lang
+     * @param rows the rows
+     * @param start the start
+     * @param targetLangs the target langs
+     * @param thesaurus the thesaurus
+     * @param type the type
+     * @param uri the uri
+     * @param sort the sort
+     * @param request the request
+     * @param httpSession the http session
+     * @return the list
+     * @throws Exception the exception
+     */
+    @ApiOperation(
+        value = "Search keywords (XML)",
+        nickname = "searchKeywordsXml",
+        notes = "")
+    @RequestMapping(
+        path = {"/search", "/search.xml"},
```

Comment:
You may be able to get the response as XML by using `_content_type` URL parameter and add XML as format supported by `searchKeywords` method ? You may need to adapt the response based on the Accept header ...

URL would look like http://localhost/geonetwork/srv/api/registries/vocabularies/search?_content_type=xml

`_content_type` is configured https://github.com/geonetwork/core-geonetwork/blob/master/web/src/main/webResources/WEB-INF/spring-servlet.xml#L87

Reply:
Yup, I tested this and it works. See latest commit.

Interestingly I didn't find I needed to actually examine the Accept header, I already found that `@RequestMapping` can do that for me just like it can for the HTTP method & URL.